prover/src/trace/trace_table.rs

```diff
@@ -342,16 +360,32 @@ impl<B: StarkField> Trace for TraceTable<B> {
         &self.meta
     }
 
-    fn get(&self, register: usize, step: usize) -> B {
-        self.trace.get(register, step)
+    fn read_main_frame(&self, row_idx: usize, frame: &mut EvaluationFrame<Self::BaseField>) {
+        let next_row_idx = (row_idx + 1) % self.length();
+        self.trace.read_row_into(row_idx, frame.current_mut());
+        self.trace.read_row_into(next_row_idx, frame.next_mut());
     }
 
-    fn read_row_into(&self, step: usize, target: &mut [B]) {
-        self.trace.read_row_into(step, target);
+    fn read_aux_frame<E: FieldElement>(&self, _row_idx: usize, _frame: &mut EvaluationFrame<E>) {
+        unimplemented!("default trace table implementation does not support auxiliary segments");
```

Comment:
I think we should have a default implementation of `Trace` which supports RAPs in the library - but what it would look like requires some more thinking. So, I left this for future PRs.

Reply:
But then how do you want to handle the RAP example in this PR? As using TraceTable will panic when reaching unimplemented code.

Reply:
I was thinking for the example we could do a custom implementation of the `Trace` trait, and then use this implementation as the basis for the "default" implementation in future PRs.

Reply:
Then this requires to export `TraceLayout` struct in winterfell crate I believe.

Reply:
No - these should be re-exported. Addressed this in the latest commit.